third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1184,6 +1206,127 @@ fn parse_control_exp(context: &mut Context) -> Result<(Exp, bool), Box<Diagnosti
     Ok((exp, ends_in_block))
 }
 
+// "for (iter in lower_bound..upper_bound) loop_body" transforms into
+// let iter = lower_bound;
+// while (iter < upper_bound) {
+//     loop_body;
+//     iter = iter + 1;
+// }
+fn parse_for_loop(context: &mut Context) -> Result<(Exp, bool), Box<Diagnostic>> {
+    const FOR_IDENT: &str = "for";
+    const IN_IDENT: &str = "in";
+    let start_loc = context.tokens.start_loc();
+
+    assert!(
+        matches!(context.tokens.peek(), Tok::Identifier) && context.tokens.content() == FOR_IDENT,
+        "Syntax Error. Use syntax: for (iter in lower_bound..upper_bound) loop_body."
+    );
+    context.tokens.advance()?;
+
+    consume_token(context.tokens, Tok::LParen)?;
+    let iter = parse_identifier(context)?;
+
+    assert!(
+        matches!(context.tokens.peek(), Tok::Identifier) && context.tokens.content() == IN_IDENT,
```

Comment:
There is `consume_identifier` which does the check for the token kind and value, you can use it here and elsewhere.